lib/elastomer/client/multi_percolate.rb

```diff
@@ -0,0 +1,127 @@
+module Elastomer
+  class Client
+
+    # Execute an array of percolate actions in bulk. Results are returned in an
+    # array in the order the actions were sent.
+    #
+    # The `multi_percolate` method can be used in two ways. Without a block
+    # the method will perform an API call, and it requires a bulk request
+    # body and optional request parameters.
+    #
+    # See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-percolate.html#_multi_percolate_api
+    #
+    # body   - Request body as a String (required if a block is not given)
+    # params - Optional request parameters as a Hash
+    # block  - Passed to a MultiPercolate instance which assembles the
+    #          percolate actions into a single request.
+    #
+    # Examples
+    #
+    #   # index and type in request body
+    #   multi_percolate(request_body)
+    #
+    #   # index in URI
+    #   multi_percolate(request_body, :index => 'default-index')
+    #
+    #   # block form
+    #   multi_percolate(:index => 'default-index') do |m|
+    #     m.percolate({ :type => 'default-type' }, { :author => "pea53" })
+    #     m.count({ :type => 'type2' }, { :author => "pea53" })
+    #     ...
+    #   end
+    #
+    # Returns the response body as a Hash
+    def multi_percolate(body = nil, params = nil)
+      if block_given?
+        params, body = (body || {}), nil
+        yield mpercolate_obj = MultiPercolate.new(self, params)
+        mpercolate_obj.call
+      else
+        raise 'multi_percolate request body cannot be nil' if body.nil?
+        params ||= {}
+
+        response = self.post '/_mpercolate', params.merge(:body => body)
```

Comment:
Should support the other paths as well:
```
/index/type/_mpercolate
/index/_mpercolate
/_mpercolate
```
See https://github.com/github/elastomer-client/blob/a19ffa5ab3207ecf162f1a37e70399bad1e31eb7/lib/elastomer/client/multi_search.rb#L43